Atos20/Crate
# Nd survey submit

What’s this PR do?  
- Merge BE branch code with FE branch code.
- A user is able to complete the style survey and have their data record to the DB.
- Adds flags for when a user has completed a survey so only allow a user to take the survey once.
- Modified the data in redux store to hold a user's id and survey results to write mutation and compare if user has taken survey.
 
Where should the reviewer start?  
- git pull this branch
- code/web directory
 
How should this be manually tested?  
- cd into `code` directory -> `npm start` to start both servers (FE, BE)
- Navigate localhost:3000 / localhost:8000 if you testing GraphQL queries.
- Login with a user.
- Go to crates page.
- Subscribe to a crate.
- Finish the survey.
- Try to subscribe to a new crate without logging out to test if the true survey flag is detected for that user.
- Check DB to ensure that user has their data updated.
- Attempt with another user to ensure each user is having their separate data recorded.
 
Any background context you want to provide?  
- When you refresh a page, the isSurveyCompleted flag disappears completely for some unknown reason. Will fix this in a bug branch.
 
What are the relevant tickets?  
- fix #47 
 
Screenshots (if appropriate) 
- n/a 
 
Questions: 
- n/a
